### PR TITLE
API V8

### PR DESCRIPTION
**Pull request by rastajpa**
- API change for https://api.bitcore.io/api/...
- Support for BCH testnet.
- Integration with the new Insight

Tested on:
- BTC mainnet (1-1)
- BTC testnet (1-1)
- BCH mainnet (2-3)
- BCH testnet (1-1)

### CODE REVIEW

File: src/app/services/recovery.service.ts [hunk 1]
```diff
@@ -379,65 +385,55 @@ export class RecoveryService {
...
-                return cb(null, addressData);
-              }
-              return cb();
-            }, 1000);
-          });
-        });
-      });
+    this.getAddressInfo(address.addressObject, coin, network).subscribe((res) => {
+      const addr = coin === 'bch' && this.useCashAddr
+        ? address.addressObject.toCashAddress().split(':')[1]
```
Comment (matiu) [inline src/app/services/recovery.service.ts]: in bitcore-lib 8+ `toString` is = `toCashAddress`
so you can use `toString`.

also, toString(true) will trim the prefix.

so you can use:
`const addr = address.addressObject.toString(true)`
without the `if`

File: src/app/services/recovery.service.ts [hunk 2]
```diff
@@ -379,65 +385,55 @@ export class RecoveryService {
...
+      const addressData = {
+        address: addr,
+        balance: res.balance.balance,
+        unconfirmedBalance: res.balance.unconfirmed,
+        utxo: res.utxos,
+        privKeys: address.privKeys,
+        pubKeys: address.pubKeys,
+        info: address.info,
+        index: address.index,
+        isActive: res.balance.unconfirmed + res.balance.confirmed > 0,
```
Comment (matiu) [inline src/app/services/recovery.service.ts]: an address could be active, even if it has not  unconfirmed or confirmed balance.

You need to check spent TXO.

File: src/app/services/recovery.service.ts [hunk 3]
```diff
@@ -379,65 +385,55 @@ export class RecoveryService {
   }
 
   private getAddressData(address: any, coin: string, network: string, cb: Function): any {
```
Comment (matiu) [inline src/app/services/recovery.service.ts]: please change `getAddressData` and `getAddressInfo` for better names.

File: src/app/services/recovery.service.ts [hunk 4]
```diff
@@ -379,65 +385,55 @@ export class RecoveryService {
...
+          return cb(null, addressData);
+        }
+        return cb();
+      }, 1000);
     });
   }
 
-  private checkAddress(address: any, coin: string, network: string): Promise<any> {
+  private getAddressInfo(address: any, coin: string, network: string): Observable<any> {
+    return Observable.forkJoin(
```
[the remark this answers is not in the document]
Comment (rastajpa) [reply src/app/services/recovery.service.ts]: You're right! 
`getAddressTxos()` is the new function that get txos. The balance is calculated from unspent txos (spentHeight = -2)

File: src/app/services/recovery.service.ts (hunk 1 above)
Comment (rastajpa) [reply src/app/services/recovery.service.ts]: Done! thanks!

File: src/app/services/recovery.service.ts (hunk 2 above)
Comment (rastajpa) [reply src/app/services/recovery.service.ts]: Now is active if an addr has at least 1 txo

File: src/app/services/recovery.service.ts (hunk 3 above)
Comment (rastajpa) [reply src/app/services/recovery.service.ts]: `checkAddress()` and `getAddressTxos()` are the new names

**Comment by rastajpa** [comment]
@matiu Thanks for the review, all the changes are in the last commit

**Comment by matiu** [comment]
Great job!

**Pull request merged**